### ISSUE TITLE
Cannot add more than one click point to point layer

### ISSUE THREAD

**Issue by ptlzon** [SchmollerLab/Cell_ACDC#482]
### Description:

I want to add click points and then use those click points to update segment objects in segment anything model SAM.

However, after selecting Add points with mouse clicks from Add points layer dialog, I could add only one point.
Any followed clicks did not add any points, even if I open Add points layer dialog and add another colored point type.

### Way to reproduce:

Select add points with mouse clicks in Add points layer dialog.
![image](https://github.com/SchmollerLab/Cell_ACDC/assets/157567305/14d2a145-3ebc-47aa-bf04-b26a7197b5e9)

After that, clicking anywhere, only the first click produces a point. Any later clicks did not show anything.

### Terminal output:

```Shell
20-02-2024, 08:48:18 - cellacdc-logger-acdc_gui - INFO:
------------------------
Initializing GUI v1.4.21
------------------------

20-02-2024, 08:48:18 - cellacdc-logger-acdc_gui - INFO:
------------------------
Initiliazing profilers...
------------------------

20-02-2024, 08:48:18 - cellacdc-logger-acdc_gui - INFO:
------------------------
GUI ready.
------------------------

20-02-2024, 08:49:49 - cellacdc-logger-acdc_gui - INFO:
------------------------
Asking to select a folder path...
------------------------

20-02-2024, 08:49:51 - cellacdc-logger-acdc_gui - INFO:
------------------------
Loading from D:/tmp/grayscale/20240220_082140_acdc/Images
------------------------

20-02-2024, 08:49:51 - cellacdc-logger-acdc_gui - INFO:
------------------------
Using .tif file (D:/tmp/grayscale/20240220_082140_acdc/Images\9Z15003-1_crop.tif)...
------------------------

20-02-2024, 08:49:51 - cellacdc-logger-acdc_gui - INFO:
------------------------
Reading  channel metadata...
------------------------

20-02-2024, 08:49:51 - cellacdc-logger-acdc_gui - INFO:
------------------------
Lazy loader paused.
------------------------

20-02-2024, 08:49:51 - cellacdc-logger-acdc_gui - INFO:
------------------------
Closing lazy loader...
------------------------

20-02-2024, 08:49:52 - cellacdc-logger-acdc_gui - INFO:
------------------------
Segmentation filename: D:/tmp/grayscale/20240220_082140_acdc/Images\9Z15003-1_crop_segm.npz
------------------------

20-02-2024, 08:49:54 - cellacdc-logger-acdc_gui - INFO:
------------------------
Loading 20240220_082140_acdc\Images\9Z15003-1_crop.tif...
------------------------

20-02-2024, 08:49:54 - cellacdc-logger-acdc_gui - INFO:
------------------------
Loaded paths:
Segmentation file name: 9Z15003-1_crop_segm.npz
ACDC output file name 9Z15003-1_crop__acdc_output.csv
------------------------

20-02-2024, 08:49:55 - cellacdc-logger-acdc_gui - INFO:
------------------------
Full dataset shape = Not Loaded
------------------------

20-02-2024, 08:49:55 - cellacdc-logger-acdc_gui - INFO:
------------------------
Loaded dataset shape = (1, 424, 567)
------------------------

20-02-2024, 08:49:55 - cellacdc-logger-acdc_gui - INFO:
------------------------
Number of frames = 1
------------------------

20-02-2024, 08:49:55 - cellacdc-logger-acdc_gui - INFO:
------------------------
Number of z-slices per frame = 1
------------------------

20-02-2024, 08:49:55 - cellacdc-logger-acdc_gui - INFO:
------------------------
Creating graphical items...
------------------------

20-02-2024, 08:49:56 - cellacdc-logger-acdc_gui - INFO:
------------------------
Initializing measurements...
------------------------

20-02-2024, 08:49:56 - cellacdc-logger-acdc_gui - INFO:
------------------------
Setting GUI mode to "Snapshots"...
------------------------

20-02-2024, 08:49:56 - cellacdc-logger-acdc_gui - INFO:
------------------------
20240220_082140_acdc || Basename: 9Z15003-1_crop || Segmented channel:  || Segmentation file name: _segm.npz
------------------------

20-02-2024, 08:49:56 - cellacdc-logger-acdc_gui - INFO:
------------------------
Autosaving worker started.
------------------------

20-02-2024, 08:50:13 - cellacdc-logger-acdc_gui - INFO:
------------------------
[WARNING]: ID 1 does not exist (add points by clicking)
------------------------

20-02-2024, 08:50:18 - cellacdc-logger-acdc_gui - INFO:
------------------------
[WARNING]: ID 1 does not exist (add points by clicking)
------------------------

20-02-2024, 08:50:37 - cellacdc-logger-acdc_gui - INFO:
------------------------
Closing autosaving worker...
------------------------

20-02-2024, 08:50:37 - cellacdc-logger-acdc_gui - INFO:
------------------------
Autosaving worker closed.
------------------------

20-02-2024, 08:50:38 - cellacdc-logger-acdc_gui - INFO:
------------------------
Clearing memory...
------------------------

20-02-2024, 08:50:38 - cellacdc-logger-acdc_gui - INFO:
------------------------
Closing GUI logger...
------------------------
```

### Version information:

```Shell
(acdc) D:\tmp>python -c "import sys; print(sys.version); import platform; print(platform.platform()); import cellacdc; print(f'Cell-ACDC version: {cellacdc.__version__}')"
3.9.18 | packaged by conda-forge | (main, Dec 23 2023, 16:29:04) [MSC v.1929 64 bit (AMD64)]
Windows-10-10.0.19045-SP0
Cell-ACDC version: 1.4.21
```

**Comment by ElpadoCan** [comment]
Hi @ptlzon, sorry about this one, it was a bug in adding points in 2D data. 

I fixed it now and I released `v1.4.23`. To upgrade activate the `acdc` environment and run `python -m pip install --upgrade cellacdc`. 

Since you mentioned that you want to use SAM with input points I added that feature too (in previous versions it was possible to run SAM only in automatic mode without prompts). 

The idea is to first click on points with the `Add points by clicking` function that you already tried with the difference that you will also need to be careful with the id of the points. If you click the mouse's left button you will add a new object per point. If you want to add more points for the same object, click with the right button. You will see the id of the point just beside the actual symbol of the point. 

Next, you save these points to a file with the save button. Finally, you will be able to input the end-name of the table to SAM as a parameter like in this screenshot:

![image](https://github.com/SchmollerLab/Cell_ACDC/assets/55240925/d1e16a84-cd87-4d27-9040-a47294d028c5)
 

Let me know if it works, thanks!

**Comment by ptlzon** [comment]
Hi @ElpadoCan 

Thanks for the quick fix. SAM works as you describe now.
However, I am not quite sure how to input both positive points and negative points to SAM.
I created two point layers (using only right clicks so as not to create additional IDs) and I expect them to be positive and negative prompts for SAM.
However, when I fed the point click list to SAM like above, the results seemed to only consider points layer number two.

Should I create another issue. 
If the above is not clear, I will provide more details with example screenshot later. I'm sorry, this week I'm occupied.

Your dedication to improving this FREE tool is awesome. Thanks a lot.

Lam

**Comment by ElpadoCan** [comment]
Hi @ptlzon, 

> Your dedication to improving this FREE tool is awesome. Thanks a lot.

No problem at all, I'm happy if Cell-ACDC can be useful! 

> I created two point layers (using only right clicks so as not to create additional IDs) and I expect them to be positive and negative prompts for SAM.

Each point layer will result in a separate table and since you can provide a single table to SAM only one will be considered. You need to create a single points layer including negative points. To add negative points to the same layer you need to give them an `id = 0`. In the top toolbar, you will be able to manually enter the `id` of the points (see screenshot below). Enter 0 and then use the right-click to add negative points. As soon as you click with the left button the `id` will be increased by 1. So for example you could enter 0, right-click on negative points and once you are done you click the left button on the first positive point and then keep using right-click to add positive points with the same id.

TL;DR Negative points have `id = 0`. 

In the future version, this will be documented better and I will also add the middle click (the scrolling wheel on Windows and Cmd+click on macOS) to add negative points. Let me know if it works, thanks! 

Best,
Francesco

**Comment by ptlzon** [comment]
Hi @ElpadoCan 

Thanks for the clarification.
I could use negative and positive points now with your instruction.
However, it seems that for my case, the combination of positive/negative points did not work as I expect (I was hoping it to produce much nicer output) or I need to spend more time understanding how to use it.

Thanks a lot.
Sincerely,
Lam

**Comment by ElpadoCan** [comment]
I'm glad to heart it "works". I guess how well it works it comes down to the dataset but if you see any suspicious behaviour feel free to report it.

Also, if you want to discuss dataset specific issues feel free to open a discussion on the image.sc forum https://forum.image.sc/

Cheers
Francesco
